### ISSUE TITLE
[package] abseil/20210324.2: abseil_BUILD_MODULES_PATHS_RELEASE/DEBUG corrupted

### ISSUE THREAD

**Issue by sbonnalc** [conan-io/conan-center-index#10545]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **abseil/20210324.2**
  * Operating System+version: **Windows10**
  * Compiler+version: **clang12**
  * cmake version: **3.21.0**
  * Conan version: **conan 1.44.1**
  * Python version: **Python 3.9.10**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
build a simple project:
conanfile.txt:
--------------------
[requires]
abseil/20210324.2

[generators]
CMakeDeps

CMakeList.txt
-------------------
cmake_minimum_required(VERSION 3.20)
project(abseil_issue CXX)

find_package(absl)

add_executable(${PROJECT_NAME} src/main.cpp)

target_link_libraries(${PROJECT_NAME} abseil::abseil)

src/main.cpp
--------------------
#include <iostream>

int main()
{
	std::cout << "abseil-issue\n";
	return 0;
}

conan install . --install-folder out --build=missing -s compiler="Visual Studio" -s compiler.version=16
Then in the out/absl-release-x86_64-data.cmake, you can check the abseil_BUILD_MODULES_PATHS_RELEASE variable is really weird:
set(abseil_BUILD_MODULES_PATHS_RELEASE "${abseil_PACKAGE_FOLDER_RELEASE}/C"
            "${abseil_PACKAGE_FOLDER_RELEASE}/:"
            "/"
            "${abseil_PACKAGE_FOLDER_RELEASE}/."
            "${abseil_PACKAGE_FOLDER_RELEASE}/c"
            ...

if you do the same with conan 1.44.1 and sup it gives this result.
But, if you use conan 1.44.0 the absl-release-x86_64-data.cmake file is ok
You can also with conan 1.44.1, modify the recipes/abseil/all/conanfile.py with this [commit](https://github.com/conan-io/conan-center-index/blob/fe65d8caca6a4ae735924cf79465b773c8523f82/recipes/abseil/all/conanfile.py), and it will work.
But with the following [commit](https://github.com/conan-io/conan-center-index/blob/38e6b8b533baac7f2ac6dcc7ddbeee519589bbbc/recipes/abseil/all/conanfile.py), it doesn't work anymore.

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Put your log output here
```

</details>

**Comment by jgsogo** [comment]
Hi, @sbonnalc . If you get different behaviors with different Conan versions it is better that you report that to Conan client repository, first of all, we need to know which one is the expected/right behavior. These new generators that aim to Conan v2 suffer changes from time to time (v1.44.1: https://github.com/conan-io/conan/pull/10326)

After that PR merged to 1.44.1 (I didn't knew about it), probably we should change these lines in the recipe:

```
        cxx_std_build_module = os.path.join(self.package_folder, self._module_path, self._cxx_std_build_module)
        self.cpp_info.set_property("cmake_build_modules", cxx_std_build_module)
        self.cpp_info.components["absl_config"].build_modules["cmake_find_package"] = [cxx_std_build_module]
        self.cpp_info.components["absl_config"].build_modules["cmake_find_package_multi"] = [cxx_std_build_module]
```

and use `cmake_build_modules` at component level.

**Comment by sbonnalc** [comment]
Do you want me to test those modifications on my test case?
